### PR TITLE
[AltCensoredChannelIE] Fix page_count regular expression

### PR DESCRIPTION
**Pull request by marcdumais**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

The current `page_count` regexp does not take into account that `channel_id` may contain one or more dash "-" character(s). If a channel with such an id is downloaded, the detection of the number of pages fails and the default value "1" is used, resulting in only the first page being downloaded.

A `ChannelIE` download test was added to cover the scenario above. Also fixed a couple of failing existing tests in this extractor.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/altcensored.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ class AltCensoredIE(InfoExtractor):
             'duration': 926.09,
             'thumbnail': 'https://archive.org/download/youtube-k0srjLSkga8/youtube-k0srjLSkga8.thumbs/k0srjLSkga8_000925.jpg',
             'view_count': int,
-            'categories': ['News & Politics'],
+            'categories': ['News', '&', 'Politics'],
```
Comment (bashonly) [inline yt_dlp/extractor/altcensored.py:35]: this is wrong and the cause of it should be fixed in a separate PR/commit
```suggestion
            'categories': ['News & Politics'],  # FIXME
```

**Comment by bashonly** [comment]
Thanks for the PR. Can you check the box to allow edits by maintainers, please?

**Comment by marcdumais** [comment]
> Can you check the box to allow edits by maintainers, please?

done

**Review by bashonly — APPROVED** [review]
